pkg/tags/tags_test.go

```diff
@@ -52,3 +59,57 @@ func TestAll(t *testing.T) {
 		t.Fatalf("expected length to be 3 got %d", len(all))
 	}
 }
+
+func TestPersistence(t *testing.T) {
+	mockStatestore := statestore.NewStateStore()
+	logger := logging.New(ioutil.Discard, 0)
+	ts := NewTags(mockStatestore, logger)
+	ta, err := ts.Create("one", 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ta.Total = 10
+	ta.Seen = 2
+	ta.Split = 10
+	ta.Stored = 8
+	ta.DoneSplit(swarm.ZeroAddress)
+	ts.Delete(ta.Uid) // simulate node going down
+
+	rcvd1, err := ts.Get(ta.Uid)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if ta.Uid != rcvd1.Uid {
+		t.Fatalf("invalid uid: expected %d got %d", ta.Uid, rcvd1.Uid)
+	}
+	if ta.Total != rcvd1.Total {
+		t.Fatalf("invalid total: expected %d got %d", ta.Total, rcvd1.Total)
+	}
+
+	// See if tag update hapens after syncing is over
+	for i := 0; i < 8; i++ {
+		ta.Inc(StateSent)
+		ta.Inc(StateSynced)
+	}
+	ts.Delete(ta.Uid) // simulate node going down
```

Comment:
Instead to use Delete, `ts` should be closed and a new one should be created with the same storage. This is test is not testing the persistence without the actual closing of the tags.